bindings/bind_py_structure_manager.cc

```diff
@@ -458,6 +458,132 @@ namespace rascal {
         py::call_guard<py::gil_scoped_release>());
   }
 
+  /**
+   * Bind a getter for the feature matrix comming from BlockSparseProperty.
+   *
+   * it will return a dictionary with keys->atomic number vals->feature mat
+   * the feature matrices size is (n_centers, inner_size)
+   */
+  template <class Calculator, class ManagerCollection_t,
+            class ManagerCollectionBinder>
+  void bind_sparse_feature_matrix_getter(
+      ManagerCollectionBinder & manager_collection) {
+    manager_collection.def(
+        "get_sparse_feature_matrix",
+        [](ManagerCollection_t & managers, Calculator & calculator) {
+          using Manager_t = typename ManagerCollection_t::Manager_t;
+          using Prop_t = typename Calculator::template Property_t<Manager_t>;
+          using Keys_t = typename Prop_t::Keys_t;
+          using VecMap_t = Eigen::Map<math::Vector_t>;
```

Comment:
I prefer it yes

Reply:
My problem with it is that when reading the code I have to jump back just to understand what is the concrete type of a given variable, which make it harder to understand. Having `Eigen::Map<math::Vector_t>` as a type mean I need to keep less context in my head

Reply:
I understand your point but I find it even more irritating to have things like `typename Prop_t::Keys_t` or `Eigen::Map<math::Vector_t>`. Also the number of alias and the length of the function is not so large right ?

Reply:
> I understand your point but I find it even more irritating to have things like typename Prop_t::Keys_t or Eigen::Map<math::Vector_t>. 

This boils down to optimizing code for reading it or writing it. Since code is written once but read multiple time by multiple people, I would rather optimize for reading it.

Plus, `auto` should help a lot here, removing the need to spell out the same complex type name multiple time.

Reply:
That's probably where we disagree: I find it more readable to have aliases than winding names. So no it does not boil down to writing optimization to me.

Reply:
I agree that **well named** aliases can make the code easier to read locally **if you remember what they refer to**. My problem is more with remember what aliases mean when they are a lot of them. 

I will not block this PR on this point, but I still think it is a worthwhile discussion to have.

Reply:
Let's discuss it next meeting then.

Reply:
👍